x-pack/plugins/ml/public/application/components/saved_objects_warning/saved_objects_warning.tsx

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FC, useEffect, useState } from 'react';
+
+import { EuiCallOut, EuiLink, EuiSpacer } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { JobType } from '../../../../common/types/saved_objects';
+import { useMlApiContext, useMlKibana } from '../../contexts/kibana';
+
+interface Props {
+  jobType?: JobType;
+}
+
+export const SavedObjectsWarning: FC<Props> = ({ jobType }) => {
+  const {
+    savedObjects: { initSavedObjects },
+  } = useMlApiContext();
+  const {
+    services: {
+      http: { basePath },
+    },
+  } = useMlKibana();
+
+  const [showWarning, setShowWarning] = useState(false);
+
+  useEffect(() => {
+    initSavedObjects(true)
+      .then(({ jobs }) => {
+        const missingJobs = jobType === undefined || jobs.some(({ type }) => type === jobType);
+        setShowWarning(missingJobs);
```

Comment:
this effect potentially can update the state of an unmounted component

Reply:
Isn't this the case for all async calls that are made from a `useEffect`?
Normally we call an async function but do not await on it as you can't pass an async function to `useEffect`, here I decided to use a `then` rather than create a second function.
This is a pattern i've seen in plenty of places, is there a better alternative?

Reply:
Check out this example https://github.com/elastic/kibana/blob/7a7057eba7270042a230ff4ac4f2404145357312/x-pack/plugins/ml/public/application/timeseriesexplorer/components/timeseries_chart/timeseries_chart_with_tooltip.tsx#L112 , I don't know if there is a better way

Reply:
ok yeah, a lock like this is a good solution.
We have a **lot** of components which have this potential problem.

Reply:
updated in https://github.com/elastic/kibana/pull/85794/commits/4eedc5485ad97546dc7cb460868e10aedd6a41b8

Reply:
yes, you're right, I see a warning message from React on this issue quite often.